packages/chain-bridge/src/router/BridgeRouter.sol

```diff
@@ -0,0 +1,828 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.28;
+
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {SafeERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import {IBridgeRouter} from "../interfaces/IBridgeRouter.sol";
+import {IBridgeAdapter} from "../interfaces/IBridgeAdapter.sol";
+import {BridgeTypes} from "../libraries/BridgeTypes.sol";
+import {EnumerableSet} from "@openzeppelin/contracts/utils/structs/EnumerableSet.sol";
+import {ProtocolAccessManaged} from "@summerfi/access-contracts/contracts/ProtocolAccessManaged.sol";
+import {ISendAdapter} from "../interfaces/ISendAdapter.sol";
+import {ReentrancyGuard} from "@openzeppelin/contracts/utils/ReentrancyGuard.sol";
+import {ICrossChainStateReadReceiver} from "../interfaces/ICrossChainStateReadReceiver.sol";
+import {IERC165} from "@openzeppelin/contracts/interfaces/IERC165.sol";
+import {ICrossChainArk} from "../interfaces/ICrossChainArk.sol";
+import {IInflightAssetTracking} from "../interfaces/IInflightAssetTracking.sol";
+import {Nonces} from "@openzeppelin/contracts/utils/Nonces.sol";
+
+/**
+ * @title BridgeRouter
+ * @notice Central router that coordinates cross-chain asset transfers and data queries
+ * @dev Implements IBridgeRouter interface and manages multiple bridge adapters.
+ *      Operations can only be initiated via the BridgeQueue or governance.
+ */
+contract BridgeRouter is
+    IBridgeRouter,
+    ProtocolAccessManaged,
+    ReentrancyGuard,
+    Nonces
+{
+    using SafeERC20 for IERC20;
+    using EnumerableSet for EnumerableSet.AddressSet;
+
+    /*//////////////////////////////////////////////////////////////
+                            STATE VARIABLES
+    //////////////////////////////////////////////////////////////*/
+
+    /// @notice Set of registered adapters
+    EnumerableSet.AddressSet private adapters;
+
+    /// @notice Mapping of operation IDs to their current status
+    mapping(bytes32 operationId => BridgeTypes.OperationStatus status)
+        public operationStatuses;
```

Comment:
the operationStatuses in bridgeRouter seems to overlap with statuses in the bridge queue, and they are inconsitent ( to be verified ). In queue tey are either queued or sent - and FAILED if they get dequeued - but this seems unnecessary - and not needed for the onchain audit trail-they could jsut be removed from the queue.


https://github.com/OasisDEX/summer-earn-protocol/blob/41dec40c3a2179f6015fd0e98aab5f56a732050a/packages/chain-bridge/src/router/BridgeQueue.sol#L84-L85